kalite/distributed/static/js/distributed/exercises/views.js

```diff
@@ -337,22 +337,34 @@ window.ExerciseView = Backbone.View.extend({
 
         clear_messages();
 
-        item.fetch({
-            success: self.render_perseus_exercise,
-            error: function() {
-                self.get_assessment_item(attempts+1);
-            }});
+
+        // Do this in this way, so that if the view is closed prior to the fetch completing
+        // successfully, the listens will have been unbound, and the callbacks will not get
+        // called.
+        this.listenToOnce(item, "sync", self.render_perseus_exercise);
+        this.listenToOnce(item, "error", function() {
+            self.get_assessment_item(attempts+1);
+        });
+
+        item.fetch();
```

Comment:
Yes, great call. This is probably the way we should always be doing things, rather than using the XMLHttpRequest callbacks.